src/main/java/com/direwolf20/buildinggadgets/common/items/ITemplate.java

```diff
@@ -73,4 +90,9 @@ default BlockPos getEndPos(ItemStack stack) {
         return GadgetUtils.getPOSFromNBT(stack, NBTKeys.GADGET_END_POS);
     }
 
+    @OnlyIn(Dist.CLIENT)
+    default void openGui() {
```

Comment:
Could this be renamed to something like open MaterialList?

Also I don't like this extension pattern - in my opinion it should directly call that...